### PR TITLE
Eliminate CSS warnings from build process

### PR DESCRIPTION
**Pull request by stevector**
The volume of warnings within our build process related to CSS order is problematic. Here's an example:

```
warning undefined
chunk styles [mini-css-extract-plugin]
Conflicting order. Following module has been added:
 * css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[2]!./src/layout/header/style.css
despite it was not able to fulfill desired ordering with these modules:
 * css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[2]!./src/components/headerBody/style.css
   - couldn't fulfill desired order of chunk group(s) component---src-pages-glossary-js
   - while fulfilling desired order of chunk group(s) component---src-templates-certificationpage-js, component---src-templates-doc-js, component---src-templates-guide-js, component---src-templates-terminus-command-js, component---src-templates-terminuspage-js, component---src-templates-video-js
 * css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[2]!./src/components/contributorGuest/style.css
   - couldn't fulfill desired order of chunk group(s) component---src-pages-glossary-js
   - while fulfilling desired order of chunk group(s) component---src-templates-certificationpage-js, component---src-templates-doc-js, component---src-templates-guide-js, component---src-templates-terminus-command-js, component---src-templates-terminuspage-js, component---src-templates-video-js
 * css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[2]!./src/components/twitter/style.css
   - couldn't fulfill desired order of chunk group(s) component---src-pages-glossary-js
   - while fulfilling desired order of chunk group(s) component---src-templates-certificationpage-js, component---src-templates-doc-js, component---src-templates-guide-js, component---src-templates-terminus-command-js, component---src-templates-terminuspage-js, component---src-templates-video-js
 * css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[10].oneOf[1].use[2]!./src/components/github/style.css
   - couldn't fulfill desired order of chunk group(s) component---src-pages-glossary-js
   - while fulfilling desired order of chunk group(s) component---src-templates-certificationpage-js, component---src-templates-doc-js, component---src-templates-guide-js, component---src-templates-terminus-command-js, component---src-templates-terminuspage-js, component---src-templates-video-js
```

In the latest build of the main branch there are 16 instances of `Conflicting order. Following module has been added:`

It is not clear to me how serious these warnings are in and of themselves, but I think the are worth eliminating purely to get rid of the noise that makes it much harder to troubleshoot other warnings and errors with the build process when they occur.

In researching this pattern of warnings I found that the path to resolution seemed to be in ensuring that components/css files were imported in the same order across pages (and potentially imported in the order of usage within a component).

In any case, the build of this PR contains no CSS warnings. The changes in the PR fall into three categories
* Reordering or `import` statements within component files to create consistency across components.
* Abstracting usage of MDX to DRY out the codebase and reduce the likelihood that the long lists import statements in the pre-existing unDRY codebase cause conflicting ordering.
* Removal of unused `import` statements. As I was cleaning up these files I was relying on VSCode's hightlighting to point out unused `import` statements. Some of the removed imports might have been irrelevant to the css ordering errors, but it seemed wise to remove unused imports anyway.

### CODE REVIEW

**Comment by stevector** [comment]
@mel-miller This PR is not at all urgent. Just some clean up that should get your review.

**Review by mel-miller — APPROVED** [review]
Works for me.

**Comment by stevector** [comment]
I just resolved the merge conflicts with `main`

**Comment by rachelwhitton** [comment]
@stevector I added local VRT here in this PR, can you take a look? On another branch I got 8 scenarios passed (screenshot of that has been added to the description at the top of this PR), but on this branch the test result is showing me 4 pages and 4 failed 

screencapture-file-Users-rachelwhitton-documentation-backstop-data-html-report-index-html-2024-05-09-13_05_39

**Review by rachelwhitton — CHANGES_REQUESTED** [review]
@stevector I added local VRT here in this PR, can you take a look? On another branch I got 8 scenarios passed (screenshot of that has been added to the description at the top of this PR), but on this branch the test result is showing me 4 passed and 4 failed 

The TOC stopped working across various templates, for example: 
* https://pr-8968-documentation.appa.pantheon.site/guides/mariadb-mysql/kill-mysql-queries
* https://pr-8968-documentation.appa.pantheon.site/certification/study-guide
* https://pr-8968-documentation.appa.pantheon.site/drupal-cron

Headers are smaller in the preview environment compared to production 

screencapture-file-Users-rachelwhitton-documentation-backstop-data-html-report-index-html-2024-05-09-13_05_39

File: backstop.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{
+  "viewports": [
+    {
+      "label": "desktop",
+      "width": 1280,
+      "height": 1080
+    }
+  ],
+  "onBeforeScript": "puppet/onBefore.js",
+  "onReadyScript": "puppet/onReady.js",
```
Comment (rachelwhitton) [inline backstop.json:10]: These 2 lines were copied in from an example config file, I think we can remove?  

```
CREATING NEW REFERENCE FILE
WARNING: script not found: /Users/rachelwhitton/documentation/backstop_data/engine_scripts/puppet/onBefore.js
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Browser Console Log 0: JSHandle:BackstopTools have been installed.
WARNING: script not found: /Users/rachelwhitton/documentation/backstop_data/engine_scripts/puppet/onReady.js
```

**Comment by stevector** [comment]
> The TOC stopped working across various templates

@rachelwhitton I'm reassigning to myself to investigate that

**Comment by stevector** [comment]
@rachelwhitton I figure out why the Table of Contents stopped working.

In the PR that created Release Notes (#8802) I created the `MdxWrapper` component. In creating that component I tried to consolidate as many of the `shortcodes` passed in to mdx from other templates including this one from the old `changelogs` template. https://github.com/pantheon-systems/documentation/blob/87ec5f58079b9876861564984b07da632a837bf8/src/templates/changelogs.js#L23

That `releaseHeadlines` component appears to exist only to _exclude_ headings from TOCs (because) on the old changelog you won't want every heading on the TOC, just the main headline (the month).

Anyway, I don't think we need the `releaseHeadlines` component at all anymore.

**Comment by stevector** [comment]
I'll keep this PR assigned to myself to continue poking at VRT, at least until our 1:1 tomorrow.

**Comment by stevector** [comment]
> Anyway, I don't think we need the releaseHeadlines component at all anymore

🙃 No, it's used elsewhere so I'll keep the component, just remove this particular usage.

**Comment by rachelwhitton** [comment]
I removed backstop from this PR and merged main then ran VRT locally and 0 passed 

screencapture-localhost-9323-2024-06-11-11_59_14

**Comment by stevector** [comment]
@rachelwhitton, I think this PR is ready to merge now. I ran VRT locally and just got some false positives

**Review by rachelwhitton — APPROVED** [review]

**Pull request merged**